manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java

```diff
@@ -225,6 +243,24 @@ public Processor updateProcessor(String bridgeId, String processorId, String cus
         return existingProcessor;
     }
 
+    private <T extends Gateway> T unmaskRequestedGateway(T requestedGateway, T existingGateway) {
```

Comment:
@kostola Just wondering if this method is better placed on the `GatewaySecretsHandler` class instead, to centralise the mask/unmask logic.

Reply:
I renamed this method because the name was misleading. This is not actually an unmasking, because the definition in this PR is always unmasked in the database. For this reason I don't think it's useful to add an `unmask` method to `GatewaySecretsHandler` at the moment (it is required in the next PR where I will interact with the AWS Secrets Manager).
What I am doing here is merging the existing secrets values for fields that are unchanged inside the request, so that the following checks perform correctly (e.g. if the processor is unchanged nothing is actually done). Without the merge, the checks would always fail because the unchanged secrets values are missing in the request.